src/Telemetry.cpp

```diff
@@ -0,0 +1,243 @@
+#include "Telemetry.h"
+
+#include <algorithm>
+#include <fstream>
+#include <iostream>
+
+#include "IRMutator.h"
+#include "Util.h"
+
+namespace Halide {
+namespace Internal {
+
+namespace {
+
+// TODO: for now we are just going to ignore potential issues with
+// static-initialization-order-fiasco, as Telemetry isn't currently used
+// from any static-initialization execution scope.
+std::unique_ptr<Telemetry> active_telemetry;
+
+class AnonymizeNames : public IRMutator {
+    using IRMutator::visit;
+
+    std::map<std::string, std::string> remapping;
+
+    // TODO: is this the right way to anonymize extern calls and image input names?
+    Expr visit(const Call *op) override {
+        if (op->call_type == Call::Extern || op->call_type == Call::ExternCPlusPlus) {
+            return Variable::make(op->type, remap(op->name, "define_extern_"));
+        } else if (op->call_type == Call::Image) {
+            return Variable::make(op->type, remap(op->name, "image"));
```

Comment:
I think this might confuse analysis to map calls to variables. Better to leave them as calls but anonymize the name.

Reply:
Ack, I intended to do so, how embarassing